### ISSUE TITLE
VM error when fetching grid.json

### ISSUE THREAD

**Issue by heaversm** [klokantech/tileserver-php#62]
Hi - I'm getting the attached
<img width="535" alt="screen shot 2015-12-17 at 4 06 23 pm" src="https://cloud.githubusercontent.com/assets/520541/11882082/8289fa94-a4d8-11e5-95f4-b34d2bbfe4f0.png">
 console error when attempting to access my .mbtiles in javascript

The output of that json file looks like this:

<img width="692" alt="screen shot 2015-12-17 at 4 06 38 pm" src="https://cloud.githubusercontent.com/assets/520541/11882093/9482c06e-a4d8-11e5-9a90-1cb0240c5562.png">

The error appears to be tied to these lines of code:

```
          //adds legend (data) to output
          $grid .= ',"data":{';
          $result = $this->db->query('SELECT key_name as key, key_json as json FROM grid_data WHERE zoom_level=' . $z . ' and tile_column=' . $x . ' and tile_row=' . $y);
          while ($r = $result->fetch(PDO::FETCH_ASSOC)) {
            $grid .= '"' . $r['key'] . '":' . $r['json'] . ',';
          }
          $grid = rtrim($grid, ',') . '}}';
```

Any idea what's going on or how to suppress these errors?

**Comment by klokan** [comment]
Not enough information to help you, I am afraid...

Is it related to legend?
Do you have a public URL where this error is visible?

**Comment by heaversm** [comment]
Yes - the url is http://107.170.120.88/map

**Comment by heaversm** [comment]
The only way I could find a way to get rid of this error was to do this:

```
$grid .= '{"data":{';
          $result = $this->db->query('SELECT key_name as key, key_json as json FROM grid_data WHERE zoom_level=' . $z . ' and tile_column=' . $x . ' and tile_row=' . $y);
          while ($r = $result->fetch(PDO::FETCH_ASSOC)) {
            $grid .= '"' . $r['key'] . '":' . $r['json'] . ',';
          }
          $grid = rtrim($grid, ',') . '}}';
```

But I'm guessing if any data were returned for the grid, this would not append the data properly. Is there a better way to disregard the grid? I'm not using one, so I'm not sure why the mbtileset would be including one (or failing the check).

**Comment by daliborjanak** [comment]
It was actualy bug in new versions of PHP related with change behavior of PDO response. We also have it, see: http://cynefin.archiveswales.org.uk/tileserver/cynefin-parishes/2/3/1.grid.json but our clients don't request grid tiles out of boundingbox so we could not detect this. 

I found it and fix it together with another bug (f5408fd10893c2e677c48e7a4af3a47d21a35e4b) related with last hybrid tile changes. It was hard to debug two bugs together :-(. 

For future: Please turn on debuging on server. Send us your vesion of PHP etc. It will save our time.
